### PR TITLE
feat(router): add max_batch_size

### PR DESCRIPTION
**Pull request by OlivierDehaene**
Some hardware require a maximum batch size.

### CODE REVIEW

File: launcher/src/main.rs [hunk 1]
```diff
@@ -279,6 +279,11 @@ struct Args {
     #[clap(default_value = "20", long, env)]
     max_waiting_tokens: usize,
 
+    /// Enforce a maximum number of requests per batch
+    /// Specific flag for hardware targets that do not support unpadded inference
```
Comment (dacorvo) [inline launcher/src/main.rs:283]: 'unpadded inference' is a bit unclear to me, but I suppose it corresponds to other configurations.

**Review by dacorvo — APPROVED** [review]
LGTM, thanks !

**Review by Narsil — APPROVED** [review]
Look much better thana my PR

File: launcher/src/main.rs (hunk 1 above)
Comment (Narsil) [reply launcher/src/main.rs:283]: Unpadded = Flash attention "varlen" (which means requests are stacked linearly without requiring padding tokens).

Padded:

input_ids = [
 [3, 53, 63],
 [0, 0, 234],
]

Unpadded:

input_ids = [3, 53, 63, 234] + [0, 3, 4] (seqlengths)

**Pull request merged**